src/transpiler/transpiler.ts

```diff
@@ -741,6 +798,27 @@ export function transpile(
     return { transpiled: '' }
   }
   const functionsToStringMap = generateFunctionsToStringMap(program)
+
+  const gpuDisplayStatements: es.Statement[] = []
+  if (variant === 'gpu') {
+    const res = transpileToGPU(program)
+
+    // add some display statements to program
+    if (res.length > 0) {
+      for (const arr of res) {
```

Comment:
Would it be better to move the details of generating display statements to somewhere in the `gpu` folder? I'm not sure.

Reply:
I see what you mean, yes let's move it to `\gpu` folder : all GPU related logic should be housed there.